Foundation/Parser.hs

```diff
@@ -276,3 +308,76 @@ skipWhile p = Parser $ \buf err ok ->
 -- stream
 skipAll :: Sequential input => Parser input ()
 skipAll = Parser $ \buf err ok -> runParser flushAll buf err ok
+
+data Count = Never | Once | Twice | Other Int
```

Comment:
I'm really not convinced this is worth it to name special cases like Never, Once, Twice. Just replacing the whole `Count` by a `Word` seems more flexible and less polluting to the parsing namespace.